### PR TITLE
Add "Roadmap" document with an overview of projects in consideration.

### PR DESCRIPTION
**Pull request by mtias**
Looking to add a document that we can easily refer to and display containing the main areas of interest for upcoming projects.

Once this is in, we could pin an issue to refer to it.

### [View Document](https://github.com/WordPress/gutenberg/blob/ca0abb320cab31c8635e6453612d2c38e4b5cefa/docs/roadmap.md)

### CODE REVIEW

File: docs/roadmap.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+- **Multi-Block Editing** — allow modifying attributes of multiple blocks of the same kind at once.
+- **Rich Text Roadmap** — continue to develop the capabilities of the rich text package. (See overview at #13778.)
+- **Common Block Functionality** — coalesce into a preferred mechanism for creating and sharing chunks of functionality (block alignment, color tools, etc) across blocks with a simple and intuitive code interface. (Suggested exploration: React Hooks.)
+- **Responsive Images** — propose mechanisms for handling flexible image sources that can be optimized for loading and takes into account their placement on a page (within main content, a column, sidebar, etc).
+- **Async Loading** — propose a strategy for loading block code only when necessary in the editor without overhead for the developer or disrupting the user experience.
+- **Styles** — continue to develop the mechanisms for managing block style variations and other styling solutions. (See overview at #7551 and #9534.)
+- **Bundling Front-end Assets** — explore ways in which front-end styles for blocks could be assembled based on which blocks are used in a given page response. (See overview at #5445.)
```
Comment (youknowriad) [inline docs/roadmap.md]: Another good thing that needs improvements and that is technically challenging is the "Transforms API". Related issue here https://github.com/WordPress/gutenberg/issues/14755

**Review by youknowriad — COMMENTED** [review]
Thanks for starting this @mtias I think that brings a lot of clarity and it may also help triage and close the related issues that express a specific part of these big challenges.

I think "roadmap" may be misinterpreted a little bit but I don't have a better alternative.

**Comment by chrisvanpatten** [comment]
Maybe `s/Roadmap/Upcoming Initiatives/g`? And then `s/Timeline/Roadmap/g`?

Love the overview overall though!

File: docs/roadmap.md (hunk 1 above)
Comment (mtias) [reply docs/roadmap.md]: Let's add it.

Comment (youknowriad) [reply docs/roadmap.md]: done

**Review by youknowriad — APPROVED** [review]
Should we add this somewhere in the handbook?

**Comment by gziolo** [comment]
This is great, I hope it’s going to be kept up to date 😃

**Comment by nosolosw** [comment]
For wider visibility, it could be helpful to have this published in the handbook. Perhaps in the contributors section?

**Pull request merged**